src/Agent.Worker/TestResults/Legacy/TestRunPublisher.cs

```diff
@@ -147,6 +147,8 @@ public async Task EndTestRunAsync(TestRunData testRunData, int testRunId, bool p
         public TestRunData ReadResultsFromFile(TestRunContext runContext, string filePath)
         {
             Trace.Entering();
+             if(!(Path.IsPathRooted(filePath)))
```

Comment:
Since it's at two places can you put it into small private function and use it ?